x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/services/analytics_service/delete_analytics.ts

```diff
@@ -147,7 +151,14 @@ export const canDeleteIndex = async (
     if (!privilege) {
       return false;
     }
-    return privilege.securityDisabled === true || privilege.has_all_requested === true;
+    let canDelete: boolean = false;
+    if (isSecurityHasPrivilegesDisabledResponse(privilege)) {
+      canDelete = privilege.securityDisabled === true;
+    }
+    if (isSecurityHasPrivilegesResponse(privilege)) {
+      canDelete = privilege.hasPrivileges.has_all_requested === true;
+    }
+    return canDelete;
```

Comment:
using the type guards here seems unnecessary. this could be:
```
return privilege.hasPrivileges === undefined || privilege.hasPrivileges.has_all_requested === true;
```
I don't think the type guards are needed at all anymore.

Reply:
Return value of hasPrivileges simplified and need for type guards removed in 7a6436dd9010700a51c91d882bd285b9120861a5